packages/contracts-bedrock/src/L1/ProxyAdminOwnedBase.sol

```diff
@@ -1,26 +1,49 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.15;
 
-import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+// Libraries
 import { Storage } from "src/libraries/Storage.sol";
 import { Constants } from "src/libraries/Constants.sol";
 
-/// @notice Base contract for ProxyAdmin-owned contracts. It's main goal is to expose the ProxyAdmin owner address on
-///         a function and also to check if the current contract and a given proxy have the same ProxyAdmin owner.
+// Interfaces
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+
+/// @notice Base contract for ProxyAdmin-owned contracts. This contract is used to introspect
+///         compatible Proxy contracts so that their ProxyAdmin and ProxyAdmin owner addresses can
+///         be retrieved onchain. Existing Proxy contracts don't have these getters, so we need a
+///         base contract instead.
 abstract contract ProxyAdminOwnedBase {
     /// @notice Getter for the owner of the ProxyAdmin.
-    ///         The ProxyAdmin is the owner of the current proxy contract.
     function proxyAdminOwner() public view returns (address) {
+        return proxyAdmin().owner();
+    }
+
+    /// @notice Getter for the ProxyAdmin contract that owns this Proxy contract.
+    function proxyAdmin() public view returns (IProxyAdmin) {
         // Get the proxy admin address reading for the reserved slot it has on the Proxy contract.
-        IProxyAdmin proxyAdmin = IProxyAdmin(Storage.getAddress(Constants.PROXY_OWNER_ADDRESS));
-        // Return the owner of the proxy admin.
-        return proxyAdmin.owner();
+        return IProxyAdmin(Storage.getAddress(Constants.PROXY_OWNER_ADDRESS));
     }
 
-    /// @notice Checks if the ProxyAdmin owner of the current contract is the same as the ProxyAdmin owner of the given
-    ///         proxy.
+    /// @notice Checks if the ProxyAdmin owner of the current contract is the same as the
+    ///         ProxyAdmin owner of the given proxy.
     /// @param _proxy The address of the proxy to check.
     function _sameProxyAdminOwner(address _proxy) internal view returns (bool) {
         return proxyAdminOwner() == ProxyAdminOwnedBase(_proxy).proxyAdminOwner();
     }
+
+    /// @notice Modifier that reverts if the caller is not the ProxyAdmin owner.
+    modifier onlyProxyAdminOwner() {
+        if (proxyAdminOwner() != msg.sender) {
+            revert("ProxyAdminOwnedBase: only the ProxyAdmin owner can call this function");
+        }
+        _;
+    }
+
+    /// @notice Modifier that reverts if the caller is not the ProxyAdmin.
+    modifier onlyProxyAdmin() {
+        if (address(proxyAdmin()) != msg.sender) {
+            revert("ProxyAdminOwnedBase: only the ProxyAdmin can call this function");
```

Comment:
why not a custom error here?

Reply:
agreed we should use our custom error convention throughout this contract

Reply:
Was just a quick sketch, will make it a custom error